package_library/packages_windows.go

```diff
@@ -182,6 +169,162 @@ func wuaUpdates(query string) ([]PkgInfo, error) {
+	if err != nil {
+		return err
+	}
+	defer count.Clear()
+
+	updtCnt, _ := count.Value().(int32)
+
+	if updtCnt == 0 {
+		logger.Infof("No updates to install")
+		return nil
+	}
+
+	logger.Infof("%d Windows updates available", updtCnt)
+
+	var msg string
+	for i := 0; i < int(updtCnt); i++ {
+		if err := func() error {
+			updtRaw, err := updts.GetProperty("Item", i)
+			if err != nil {
+				return err
+			}
+			defer updtRaw.Clear()
+
+			updt := updtRaw.ToIDispatch()
+			defer updt.Release()
+
+			title, err := updt.GetProperty("Title")
+			if err != nil {
+				return err
+			}
+			defer title.Clear()
+
+			eula, err := updt.GetProperty("EulaAccepted")
+			if err != nil {
+				updtRaw.Clear()
+				return err
+			}
+			defer eula.Clear()
+
+			msg += fmt.Sprintf("  %s\n  - EulaAccepted: %v\n", title.Value(), eula.Value())
+			return nil
+		}(); err != nil {
+			return err
+		}
+	}
+	logger.Infof("Windows updates to be installed:\n%s", msg)
+
+	if err := DownloadWUAUpdateCollection(session, updtsRaw); err != nil {
+		return fmt.Errorf("DownloadWUAUpdates error: %v", err)
+	}
+
+	if err := InstallWUAUpdateCollection(session, updtsRaw); err != nil {
+		return fmt.Errorf("InstallWUAUpdates error: %v", err)
+	}
+	return nil
+}
+
+// DownloadWUAUpdateCollection downloads all updates in a IUpdateCollection
+func DownloadWUAUpdateCollection(session IUpdateSession, updates IUpdateCollection) error {
+	// returns IUpdateDownloader
+	// https://docs.microsoft.com/en-us/windows/desktop/api/wuapi/nn-wuapi-iupdatedownloader
+	downloaderRaw, err := session.CallMethod("CreateUpdateDownloader")
+	if err != nil {
+		return fmt.Errorf("error calling method CreateUpdateDownloader on IUpdateSession: %v", err)
+	}
+	downloader := downloaderRaw.ToIDispatch()
+	defer downloaderRaw.Clear()
+
+	if _, err := downloader.PutProperty("Updates", updates); err != nil {
+		return fmt.Errorf("error calling PutProperty Updates on IUpdateDownloader: %v", err)
+	}
+
+	logger.Infof("Downloading updates")
+	if _, err := downloader.CallMethod("Download"); err != nil {
+		return fmt.Errorf("error calling method Download on IUpdateDownloader: %v", err)
+	}
+	return nil
+}
+
+// InstallWUAUpdateCollection installs all updates in a IUpdateCollection
+func InstallWUAUpdateCollection(session IUpdateSession, updates IUpdateCollection) error {
+	// returns IUpdateInstaller
+	// https://docs.microsoft.com/en-us/windows/desktop/api/wuapi/nf-wuapi-iupdatesession-createupdateinstaller
+	installerRaw, err := session.CallMethod("CreateUpdateInstaller")
+	if err != nil {
+		return fmt.Errorf("error calling method CreateUpdateInstaller on IUpdateSession: %v", err)
+	}
+	installer := installerRaw.ToIDispatch()
+	defer installerRaw.Clear()
+
+	if _, err := installer.PutProperty("Updates", updates); err != nil {
+		return fmt.Errorf("error calling PutProperty Updates on IUpdateInstaller: %v", err)
+	}
+
+	logger.Infof("Installing updates")
+	if _, err := installer.CallMethod("Install"); err != nil {
```

Comment:
Does this return an object that can be used to track progress or logs? If so, it might be worth logging that value for now to show that we have it and will use it. It seems like that could be valuable. The same for `Download`.

Reply:
It doesn't return anything as this is the synchronous method (that empty val is indeed empty), there is an async method we could use to track progress, it uses callback functions that I really didn't want to get into implementing right now. Added TODO as this could be worth spending time on in the future